altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -218,4 +261,77 @@ mod tests {
     assert!(test::current_task().is_some());
     assert_eq!(handle_3.tid(), Ok(test::current_task().unwrap().tid()));
   }
+
+  #[test]
+  fn only_critical_task() {
+    let _g = test::set_up();
+    test::create_and_schedule_test_task(512, Priority::Normal, "test task 1");
+    test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
+    start_scheduler();
+    for _ in 0..100 {
+      assert!(test::current_task().unwrap().priority == Priority::Normal);
+    }
+  }
+
+  #[test]
+  //No task, but scheduler is run we should just see the idle task
+  fn no_task_sched_run_sched() {
+    let _g = test::set_up();
+    start_scheduler();
+    assert_eq!(test::current_task().unwrap().priority, Priority::__Idle);
+  }
+
+  #[test]
+  //All tasks are blocked
+  fn all_tasks_blocked() {
+    let _g = test::set_up();
+    assert!(test::current_task().is_none());
+    let handle_1 = test::create_and_schedule_test_task(512, Priority::Low, "test task 1");
+    let handle_2 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
+    start_scheduler();
+    assert!(test::current_task().is_some());
+    test::current_task().unwrap().state = State::Blocked;
+    test::current_task().unwrap().delay_type = Delay::Sleep;
+    switch_context();
+    assert!(test::current_task().is_some());
+    test::current_task().unwrap().state = State::Blocked;
+    test::current_task().unwrap().delay_type = Delay::Sleep;
+
+    switch_context();
+    assert_ne!(Ok(test::current_task().unwrap().tid()), handle_1.tid());
+    assert_ne!(Ok(test::current_task().unwrap().tid()), handle_2.tid());
+  }
+
+  #[test]
+    fn scheduler_critical_always_wins() {
```

Comment:
This function and the next are not indented correctly. They should be aligned with #[test]. Also, some of the tests have comments explaining what they do and some don't.